runtime/lua/vim/ui/_utils.lua

```diff
@@ -0,0 +1,32 @@
+local utils = {}
+
+function utils.tbl_apply_defaults(original, defaults)
+  if original == nil then
+    original = {}
+  end
+
+  original = vim.deepcopy(original)
+
+  for k, v in pairs(defaults) do
+    if original[k] == nil then
+      original[k] = v
+    end
+  end
+
+  return original
+end
+
+function utils.tbl_longest_str(tbl)
+  local len = 0
+
+  for _,str in pairs(tbl) do
+    local str_len = #str
```

Comment:
isn't there `math.max`?

Reply:
there is :) (the behavior might be slightly different between PUC Lua and LuaJIT, though, at least in edge cases -- see https://github.com/LuaJIT/LuaJIT/commit/03208c8162af9cc01ca76ee1676ca79e5abe9b60)